fuse_optimizers/src/batch_optimizer.cpp

```diff
@@ -169,10 +169,7 @@ void BatchOptimizer::optimizerTimerCallback(const ros::TimerEvent& /*event*/)
   // Check if there is any pending information to be applied to the graph.
   {
     std::lock_guard<std::mutex> lock(combined_transaction_mutex_);
-    optimization_request_ = !combined_transaction_->addedConstraints().empty() ||
-                            !combined_transaction_->removedConstraints().empty() ||
-                            !combined_transaction_->addedVariables().empty() ||
-                            !combined_transaction_->removedVariables().empty();
+    optimization_request_ = !combined_transaction_->empty();
```

Comment:
Are you just looking to simplify this line? Because this line probably predates the creation of the `Transaction.empty()` method. The fixed-lag smoother uses the old definition of `Transaction.empty()` in a very similar way.
https://github.com/locusrobotics/fuse/blob/devel/fuse_optimizers/src/fixed_lag_smoother.cpp#L242

I think we should leave the `Transaction.empty()` method alone, and keep just this change to `optimization_request_`.

Unless you want to change the definition of `Transaction.empty()`. In which case, what is reason? I'm not necessarily against dropping the "involved_stamps" check, I just want to understand what is causing problems.

Reply:
Nothing was causing problems, just looking to simplify this line. `Transaction::empty()` was introduced after this line, yes.

I simply remove the check for `involved_stamps_.empty()` in `Transaction::empty()`, so it matches the checks done here, and because I don't think it's really needed.

Reply:
If it wasn't causing any issues, I'd prefer to leave the `Transaction::empty()` definition as it was. So `empty()` really does mean empty. But I'm all for changing this line to `optimization_request_ = !combined_transaction_->empty();`, even though it changes the criteria slightly. This brings it in line with the fixed-lag smoother.

Reply:
Works for me. I've just removed the commit that changed the `Transaction::empty()` definition.